libraries/chain/include/steem/chain/account_object.hpp

```diff
@@ -49,6 +49,9 @@ namespace steem { namespace chain {
          bool              can_vote = true;
          uint16_t          voting_power = STEEM_100_PERCENT;   ///< current voting power of this account, it falls after every vote
          time_point_sec    last_vote_time; ///< used to increase the voting power of this account the longer it goes without voting.
+#ifdef STEEM_ENABLE_SMT
+         time_point_sec    last_smt_vote_time; ///< same as last_vote_time but related to voting with SMT asset.
```

Comment:
Yes, SMT voting power is calculated separately for each SMT, but `last_vote_time` is calculated completely independent from voting power, only later they are both combined to produce `current_power` as can be seen in this snippet from `calculate_power_shares`:
```
   int64_t elapsed_seconds = (db.head_block_time() - last_vote_time).to_seconds();

   if( db.has_hardfork( STEEM_HARDFORK_0_11 ) )
      FC_ASSERT( elapsed_seconds >= voting_helper->GetMinimalVoteInterval(), 
                 "Can only vote once every ${sec} seconds.", ("sec", voting_helper->GetMinimalVoteInterval()) );

   int64_t regenerated_power = (STEEM_100_PERCENT * elapsed_seconds) / voting_helper->GetVoteRegenerationPeriod();
   info->current_power = std::min( int64_t(voting_power + regenerated_power), int64_t(STEEM_100_PERCENT) );
```
This however doesn't answer the question whether `last_vote_time` should occur on per SMT basis. My assumption that it should not origins from the fact that `SMT_MIN_VOTE_INTERVAL_SEC` is common to all SMTs per SMT Whitepaper and this is the only limit put on `last_vote_time`. Then again it's a weak assumption really so I'd like an answer from designer:
@theoreticalbts Should `last_smt_vote_time` be bound to account data per SMT separately or should there be a single value limiting every SMT voting for that account, as suggested by the fact that `SMT_MIN_VOTE_INTERVAL_SEC` is common to all SMTs?

Reply:
@theoreticalbts and I talked and agree that each SMT needs a vote time. The vote time is also used the regenerate voting power. Because each SMT has its own voting power, if we used a single last vote time for all SMTs, then we would need to update voting power for every SMT every vote. Having a separate time for each SMT means we can update the value lazily for each SMT only on votes using that SMT.